### ISSUE TITLE
InputComponent.group is not implicitly invoked by TextModeLayout

### ISSUE THREAD

**Issue by jsfan3** [codenameone/CodenameOne#2454]
In the following code, the InputComponents are aligned correctly (the `credentials` container is added to a `BoxLayout.y` Form):

```
        TextModeLayout tl = new TextModeLayout(2, 1);
        Container credentials = new Container(tl);
        credentials.setUIID("MyContainer01");
        TextComponent email = new TextComponent().label("Email");
        TextComponent password = new TextComponent().label("Password");
        Validator val = new Validator();
        val.addConstraint(email, RegexConstraint.validEmail());
        val.addConstraint(password, new LengthConstraint(8));
        InputComponent.group(email, password);
        credentials.add(email);
        credentials.add(password);
```

But... if I remove the `InputComponent.group(email, password);` the components are not aligned properly. [According to the Javadocs](https://www.codenameone.com/javadoc/com/codename1/ui/InputComponent.html#group-com.codename1.ui.Component...-), `InputComponent.group(email, password);` should be invoked implicitly when using TextModeLayout, so... is this a bug?

**Comment by codenameone** [comment]
I just tried this and it was aligned perfectly:

````java
Form hi = new Form("Alignment", BoxLayout.y());

TextModeLayout tl = new TextModeLayout(2, 1);
Container credentials = new Container(tl);
credentials.setUIID("MyContainer01");
TextComponent email = new TextComponent().label("Email");
TextComponent password = new TextComponent().label("Password");
Validator val = new Validator();
val.addConstraint(email, RegexConstraint.validEmail());
val.addConstraint(password, new LengthConstraint(8));
credentials.add(email);
credentials.add(password);
hi.add(credentials);

hi.show();
````

![codenameone screenshot 118](https://user-images.githubusercontent.com/2353641/41505393-4b9b803a-7211-11e8-869d-e71d0d321e16.png)

**Comment by jsfan3** [comment]
You're right. However, the following more complete code doesn't align properly, as you can see in my screenshot.

```
Form hi = new Form("Alignment", BoxLayout.y());
        
        SpanLabel introText = new SpanLabel("Intro text on multiple lines... this is a long sentence");
        
        // Login button (disabled until the user inputs validates)
        Label logo = new Label(FontImage.createMaterial(FontImage.MATERIAL_ACCOUNT_BALANCE_WALLET, "Label", 50));
        
        // Login button (disabled until the user inputs validates)
        Button loginButton = new Button("ButtonLogin", "MyButton01");
        loginButton.addActionListener(l -> {
            Log.p("loginButton pressed");
        });

        TextModeLayout tl = new TextModeLayout(2, 1);
        Container credentials = new Container(tl);
        credentials.setUIID("MyContainer01");
        TextComponent email = new TextComponent().label("Email");
        TextComponent password = new TextComponent().label("Password");
        password.getField().setDoneListener(l -> {
            if (loginButton.isEnabled()) {
                loginButton.pressed();
                loginButton.released();
            }
        });
        Validator val = new Validator();
        val.addConstraint(email, RegexConstraint.validEmail());
        val.addConstraint(password, new LengthConstraint(8));
        val.addSubmitButtons(loginButton);
        email.getField().setConstraint(TextField.EMAILADDR);
        password.getField().setConstraint(TextField.PASSWORD);
        // InputComponent.group(email, password);
        credentials.add(email);
        credentials.add(password);
        
        Button lostPassword = new Button("LostPassword");
        
        hi.add(FlowLayout.encloseCenter(introText));
        hi.add(FlowLayout.encloseCenter(logo));
        hi.add(credentials);
        hi.add(loginButton);

        hi.show();
```

![alignment](https://user-images.githubusercontent.com/1997316/41665826-3dab8264-74a9-11e8-83e2-926d20215f30.png)

**Comment by codenameone** [comment]
Open the component inspector and look at the preferred size of the Email/Password labels. If both have identical width it means the code works as expected. 

I'm guessing the email/password text fields have different preferred sizes which cause this issue as the layout takes that into consideration. If this is indeed the case we need to think about the "right thing" to do here.

**Comment by jsfan3** [comment]
Following up on the [requested preferred-size check](https://github.com/codenameone/CodenameOne/issues/2454#issuecomment-398986104): I retested my fuller example on 5 September 2026 in a fresh Codename One Maven project using **7.0.267**, without the explicit `InputComponent.group(email, password)` call.

The Email and Password labels have **identical preferred dimensions**. The two text fields also have identical preferred dimensions and the same left edge:

| Environment | Label preferred size, both labels | Text field preferred size, both fields | Field X, Email / Password |
| --- | --- | --- | --- |
| CN1 Simulator, Android / Galaxy S8 skin | 327 × 88 | 1121 × 188 | 332 / 332 |
| CN1 Simulator, iPhone X skin | 411 × 128 | 1126 × 140 | 452 / 452 |
| Physical Samsung Galaxy A20e (SM-A202F), Android 11 | 176 × 54 | 650 × 111 | 181 / 181 |

Values are CN1 pixels; preferred sizes were read on the EDT using `getPreferredW()` / `getPreferredH()`, the same getters used by the Component Inspector.

Both my fuller example and the earlier minimal example remained aligned with empty and populated fields, across three repetitions per configuration. Adding the explicit grouping call did not change the result. On the physical Android device, alignment also remained correct after entering both values through the native keyboard and dismissing it.

So I **cannot reproduce the reported misalignment in these configurations on 7.0.267**; automatic grouping is working in these tests.

One limitation: the 2018 example did not include the styles for `MyContainer01` / `MyButton01`, so these tests use the freshly generated project's theme defaults, not the original custom theme. The iPhone X result is from the JavaSE-based CN1 Simulator, **not** a native iOS Simulator or a physical iPhone.

Attached are screenshots of the fuller example without explicit grouping, plus the reproducer and detailed measurements.

<img width="1440" height="2960" alt="simulator-android.png" src="https://github.com/user-attachments/assets/e36c915d-3f37-4bd2-9f0e-d6bde96860b6" />

<img width="1125" height="2436" alt="simulator-iphone-skin.png" src="https://github.com/user-attachments/assets/15040d22-d731-462a-8d2f-08a16cff6f4d" />

<img width="720" height="1560" alt="android-device.png" src="https://github.com/user-attachments/assets/0d8a104d-71c0-406b-bf84-9a59aeacead5" />

[issue-2454-verification.zip](https://github.com/user-attachments/files/31856079/issue-2454-verification.zip)
